libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -176,6 +176,20 @@ void ExpressionCompiler::appendStateVariableAccessor(VariableDeclaration const&
 	m_context.appendJump(eth::AssemblyItem::JumpType::OutOfFunction);
 }
 
+bool ExpressionCompiler::visit(Conditional const& _condition)
+{
+	CompilerContext::LocationSetter locationSetter(m_context, _condition);
+	_condition.condition().accept(*this);
+	eth::AssemblyItem trueTag = m_context.appendConditionalJump();
+	_condition.falseExpression().accept(*this);
```

Comment:
I think a `utils().convertType(_condition.falseExpression().annotation().type, _condition.annotation().type)` does not hurt (also below).